src/ucp/am/eager_multi.c

```diff
@@ -77,17 +77,15 @@ static size_t ucp_am_eager_multi_bcopy_pack_args_first(void *dest, void *arg)
     ucp_am_hdr_t *hdr                    = dest;
     ucp_proto_multi_pack_ctx_t *pack_ctx = arg;
     ucp_request_t *req                   = pack_ctx->req;
-    size_t length, max_length;
+    size_t length;
 
     ucs_assertv(req->send.state.dt_iter.offset == 0, "offset %zu",
                 req->send.state.dt_iter.offset);
 
     ucp_am_fill_header(hdr, req);
 
-    max_length = ucs_min(ucp_am_send_req_total_size(req),
-                         pack_ctx->max_payload);
-    length     = ucp_am_eager_bcopy_pack_data(hdr + 1, req, max_length,
-                                              pack_ctx->next_iter);
+    length = ucp_am_eager_bcopy_pack(hdr + 1, req, pack_ctx->max_payload,
+                                     pack_ctx->next_iter);
```

Comment:
is it because the first fragment is always supposed to consume the whole fragment?

Reply:
We use `ucp_proto_multi_max_payload` to calculate `pack_ctx->max_payload`.
`ucp_proto_multi_max_payload` takes into account fragment size, length, and header size.
Thus, this line is redundant.

Reply:
ok